src/bm_sim/calculations.cpp

```diff
@@ -392,6 +392,40 @@ struct xor_hash {
   }
 };
 
+struct toeplitz {
+  uint32_t operator()(const char *buf, size_t len) const {
+    uint32_t final_xor_value = 0;
+    const uint32_t *input = reinterpret_cast<const uint32_t *>(buf);
+    const size_t n_word = len / 4;
+
+    for (size_t j = 0; j < n_word / 4; j++) {
+      for (uint32_t input_word = input[j]; input_word; input_word &= (input_word - 1)) {
+        int i = ffs(input_word);
+        uint32_t key_word_l = rss_key[j % rss_key_len] << (31 - i);
+        uint32_t key_word_r = rss_key[(j + 1) % rss_key_len] >> (i + 1);
+        final_xor_value ^= (key_word_l | key_word_r);
+      }
+    }
+
+    // TODO: Decide whether to require len % 4 == 0,
+    // or to handle remaining bytes in some way.
+
+    return final_xor_value;
+  }
+
+  // TODO: I'm not sure what would be the best API to configure the RSS key.
+  void update_rss_key(const uint32_t *new_rss_key, size_t new_rss_key_len) {
+    rss_key_len = new_rss_key_len;
+    rss_key = new uint32_t(rss_key_len);
+    std::memcpy(rss_key, new_rss_key, rss_key_len * sizeof(uint32_t));
```

Comment:
you have a memory leak when the key is updated. Please don't use `new` at all. Just use a `std::vector` for the key.

Reply:
Done.